src/main/kotlin/net/ccbluex/liquidbounce/features/chat/packet/PacketGsonExtension.kt

```diff
@@ -30,7 +31,7 @@ import java.lang.reflect.Type
  */
 class PacketSerializer : JsonSerializer<Packet> {
 
-    private val packetRegistry = hashMapOf<Class<out Packet>, String>()
+    private val packetRegistry = IdentityHashMap<Class<out Packet>, String>()
```

Comment:
In the case of `Class` equals and identity-equals are the same. Please revert this